src/collection.rs

```diff
@@ -28,13 +44,16 @@ impl fmt::Debug for FontRef {
     }
 }
 
+/// An iterator over `(Range<usize>, &FontRef)` pairs. Use the given font for the given range. When
```

Comment:
Probably best to reference the function that generates it, the type of the iterator is not that important (some APIs just write `impl Iterator<Item = ...>`).

Also, not 100% clear this should be public, I think `pub(crate)` might be better, but we can err on the side of exposing more for now.

Reply:
I've changed it a bit, let me know if its OK now.